src/models/game/epics.js

```diff
@@ -149,64 +52,33 @@ const replaceEpic = (action$, state$) =>
     })
   );
 
-const phase1CheckEpic = (action$, state$) =>
-  action$.pipe(
-    ofType(check.type),
-    filter(() => phase(state$.value) === 1),
-    map(advancePhase)
-  );
-
-const phase3CheckEpic = (action$, state$) =>
+const checkEpic = (action$, state$) =>
   action$.pipe(
     ofType(check.type),
-    filter(() => phase(state$.value) === 3),
     mergeMap(() => {
       const comparisonResult = compareHands(
         playerHand(state$.value),
         aiHand(state$.value)
       );
+
       const actions = [advancePhase()];
```

Comment:
If I understand correctly, you can add betReset here as well?

Reply:
we want betReset to run only on the final phase of the game, not every phase.